packages/react-instantsearch-dom/src/components/SmartSort.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { createClassNames } from '../core/utils';
+
+const cx = createClassNames('SmartSort');
+
+type ComponentProps = {
+  isSmartSorted: boolean;
+};
+
+export type SmartSortComponent = {
+  buttonTextComponent: React.FunctionComponent<ComponentProps>;
+  className?: string;
+  isVirtualReplica: boolean;
+  isSmartSorted: boolean;
+  textComponent: React.FunctionComponent<ComponentProps>;
+  refine(relevancyStrictness: number | undefined): void;
+};
+
+const SmartSort: React.FC<SmartSortComponent> = ({
+  buttonTextComponent: ButtonTextComponent,
+  className = '',
+  isVirtualReplica,
+  isSmartSorted,
+  textComponent: TextComponent,
```

Comment:
```suggestion
  className = '',
  isVirtualReplica,
  isSmartSorted,
  buttonTextComponent: ButtonTextComponent,
  textComponent: TextComponent,
```

not sure if it was sorted another way, but this way the "templates" are together

Reply:
It was alphabetically sorted but it's not a big deal as we don't do it in other files